server/form-pages/apply/risk-strength-and-need-information/prison-information/adjudications.ts

```diff
@@ -0,0 +1,76 @@
+import type { Adjudication, ApprovedPremisesApplication } from '@approved-premises/api'
+import type { DataServices, PageResponse } from '@approved-premises/ui'
+import { CallConfig } from '../../../../data/restClient'
+import { DateFormats } from '../../../../utils/dateUtils'
+import { sentenceCase } from '../../../../utils/utils'
+import TasklistPage from '../../../tasklistPage'
+import { Page } from '../../../utils/decorators'
+
+type AdjudicationsBody = {
+  adjudications: Array<
+    Omit<Adjudication, 'finding'> & {
```

Comment:
Presumably this is modifying the original Adjudication type and we can't or don't want to make this change on the main Adjudication. I'm thinking that if it can be null, should we not make this change globally.

Reply:
The significant bit here isn't `| null`, it's `/** @nullable */`. That's what gets us the schema we want, but even explicitly marking a field as nullable in the `api.yml` file doesn't get us the `/** @nullable */` comment in the generated types.

tbh, the more I think about this the less I like it - if the API changes and we get extra data back from the server, that's automatically going to be included, and will potentially cause the schema validation to fail. Either we use the "all" schema and this is moot, or we should do some preprocessing on the adjudications before including in our form data

Reply:
I've taken latter approach - rather than bodging the API types with extra declarations, there's a method mapping the data we get from the API to fixed UI types with stronger guarantees about what will be present